miio/integrations/vacuum/mijia/pro2vacuum.py

```diff
@@ -0,0 +1,308 @@
+    Point = 4
+    Remote = 5
+    Explore = 6
+    Room = 7
+    Floor = 8
+
+
+class RepeatState(FormattableEnum):
+    Off = 0
+    On = 1
+
+
+class DoorState(FormattableEnum):
+    Off = 0
+    DustBox = 1
+    WaterVolume = 2
+    TwoInOneWaterVolume = 3
+
+
+class ClothState(FormattableEnum):
+    Off = 0
+    On = 1
+
+
+class SuctionState(FormattableEnum):
+    Off = 0
+    EnergySaving = 1
+    Standard = 2
+    Turbo = 3
+
+
+class WaterState(FormattableEnum):
+    LowLevel = 0
+    MediumLevel = 1
+    HighLevel = 2
+
+
+class MopRoute(FormattableEnum):
+    BowStyle = 0
+    YStyle = 1
+
+
+class Pro2Status(DeviceStatus):
+    """Container for status reports from Mi Robot Vacuum-Mop 2 Pro."""
+
+    def __init__(self, data):
+        """Response (MIoT format) of a Mi Robot Vacuum-Mop 2 Pro (ijai.vacuum.v3)
+
+        [
+            {'did': 'battery', 'siid': 3, 'piid': 1, 'code': 0, 'value': 100},
+            {'did': 'error_code', 'siid': 2, 'piid': 2, 'code': 0, 'value': 0},
+            {'did': 'state', 'siid': 2, 'piid': 1, 'code': 0, 'value': 5},
+            {'did': 'fan_speed', 'siid': 2, 'piid': 6, 'code': 0, 'value': 1},
+            {'did': 'operating_mode', 'siid': 2, 'piid': 4, 'code': 0, 'value': 1},
+            {'did': 'mop_state', 'siid': 16, 'piid': 1, 'code': 0, 'value': 0},
+            {'did': 'water_level', 'siid': 2, 'piid': 5, 'code': 0, 'value': 2},
+            {'did': 'main_brush_life_level', 'siid': 14, 'piid': 1, 'code': 0, 'value': 99},
+            {'did': 'main_brush_time_left', 'siid': 14, 'piid': 2, 'code': 0, 'value': 17959}
+            {'did': 'side_brush_life_level', 'siid': 15, 'piid': 1, 'code': 0, 'value': 0 },
+            {'did': 'side_brush_time_left', 'siid': 15, 'piid': 2', 'code': 0, 'value': 0},
+            {'did': 'filter_life_level', 'siid': 15, 'piid': 2', 'code': 0, 'value': 0},
+            {'did': 'filter_time_left', 'siid': 15, 'piid': 2', 'code': 0, 'value': 0},
+            {'did': 'clean_area', 'siid': 9, 'piid': 1, 'code': 0, 'value': 0},
+            {'did': 'clean_time', 'siid': 9, 'piid': 2, 'code': 0, 'value': 0}
+            {'did': 'current_language', 'siid': 7, 'piid': 21, 'code': 0, 'value': 0}
+            ]"""
+        self.data = data
+
+    @property
+    def battery(self) -> int:
+        """Battery Level."""
+        return self.data["battery"]
+
+    @property
+    def error_code(self) -> int:
+        """Error code as returned by the device."""
+        return int(self.data["error_code"])
+
+    @property
+    def error(self) -> str:
+        """Human readable error description, see also :func:`error_code`."""
+        try:
+            return ERROR_CODES[self.error_code]
+        except KeyError:
+            return "Definition missing for error %s" % self.error_code
+
+    @property
+    def state(self) -> DeviceState:
+        """Vacuum Status."""
+        return DeviceState(self.data["state"])
+
+    @property
+    def fan_speed(self) -> SuctionState:
```

Comment:
Would it be better to rename `SuctionState` to something more related to fan speed?